lib/index.js

```diff
@@ -68,6 +70,22 @@ exports.global = window;
 
 Segment.prototype.initialize = function() {
   var self = this;
+
+  if (this.options.retryQueue) {
+    this._lsqueue = new Queue('segmentio', function(item, done) {
+      // Update the sentAt time each retry so the tracking-api doesn't interperet a time skew
+      item.sentAt = new Date();
+      // send
+      send(item.url, item.msg, item.headers, function(err, res) {
```

Comment:
I certainly didn't mean that we should be waiting for `n` events to queue before batching and making a call. I think it's only when there are a few events that failed in succession would you have a "batch", and it's more efficient (and reliable) to send the events in a batch. I do think that in 99% (or however often requests fail) of requests the batch would simply have a size of 1 and batching wouldn't impact debugging.

Either way I'm not against the current behaviour given this is much better than what we had before.